contrib/platform/src/com/sun/jna/platform/WindowUtils.java

```diff
@@ -1297,11 +1297,38 @@ public String getProcessFilePath(final HWND hwnd) {
                 final int length = Psapi.INSTANCE.GetModuleFileNameExW(process,
                                                                        null, filePath, filePath.length);
                 if (length == 0) {
-                    if(Kernel32.INSTANCE.GetLastError() != WinNT.ERROR_INVALID_HANDLE) {
-                        throw new Win32Exception(Kernel32.INSTANCE.GetLastError());
-                    } else {
+                    int lastError = Kernel32.INSTANCE.GetLastError();
+                    if(lastError == WinNT.ERROR_INVALID_HANDLE) {
                         // ignore invalid handles
                         return "";
+                    } else if (lastError == WinNT.ERROR_PARTIAL_COPY) {
+                        // observed on 64 platforms when the JVM running the JNA
+                        // code is 32 bit and the target process is 64 bit.
+                        // Fallback to GetProcessImageFileName. But the returned
+                        // path is in device form, so this needs to be mapped
+                        // back to the drive form
+                        final int length2 = Psapi.INSTANCE.GetProcessImageFileName(process,
+                            filePath, filePath.length);
```

Comment:
Not a part of this PR, but I'm curious about the hardcoded choice (line 1281) of 2048 for the `filePath` buffer.  (Related, the `pathPrefixBuffer` below.)  Searching the docs I see there was a `MAX_PATH` of 260 (`WinDef.MAX_PATH` in JNA) but that limitation has been lifted in recent Win10 versions, so the limit is now effectively 32767 bytes rather than 260 `TCHAR`s.  Should we be handling this rare corner case?

See [Kernel32.QueryFullProcessImageName](https://github.com/java-native-access/jna/blob/8a44e4c290aeda3a19d6cdad888cd5ccbe4375b9/contrib/platform/src/com/sun/jna/platform/win32/Kernel32Util.java#L911) which appears to take this approach (start with MAX_PATH and increment on failure).  Can we just call that here or is there a difference between ProcessImageName vs. ProcessImageFileName?

Reply:
Discussion of the path length and reasoning for the incrementing in the above method [here](https://github.com/java-native-access/jna/pull/1214#pullrequestreview-439708229)

Reply:
And rereading my comment (second code comment in the comment linked above) I'm not sure we should be calling [`GetModuleFileNameEx`](https://docs.microsoft.com/en-us/windows/win32/api/psapi/nf-psapi-getmodulefilenameexw) in the first place on line 1297:
> The GetModuleFileNameEx function is primarily designed for use by debuggers and similar applications that must extract module information from another process. If the module list in the target process is corrupted or is not yet initialized, or **if the module list changes during the function call** as a result of DLLs being loaded or unloaded, **GetModuleFileNameEx may fail or return incorrect information**.

The docs recommend:
> To retrieve the name of the main executable module for a remote process, use the GetProcessImageFileName or QueryFullProcessImageName function. **This is more efficient and more reliable than calling the GetModuleFileNameEx function with a NULL module handle.**

So rather than using this change as a backup, it should be the primary choice here.